Advanced training path: MIX AND OTP [4-6]

Euphorichuman opened the pull request.

Euphorichuman committed:
[ADD] Supervisor and application, dynamic supervisors, and ETS

luishurtado commented on Advanced/mix_test/test/mix_test_test.exs:
```diff
@@ -2,7 +2,7 @@ defmodule MIX_TESTTest do
-  test "greets the world" do
-    assert MIX_TEST.hello() == "Hello world!"
-  end
+  #test "greets the world" do
+  #  assert MIX_TEST.hello() == "Hello world!"
+  #end
```
You can get rid of this test instead of commenting it out.

luishurtado reviewed (changes requested):
1. Module names in Elixir are usually Capitalized. `MixTest` instead of `MIX_TEST.`
2. Folder names must be in snake_case. `advanced` instead of `Advanced.`

Euphorichuman committed:
[FIX] Module name, folder names, remove test

*Module name: `MIX_TEST` to `MixTest`
*Folder name: `Advanced` to `advanced`
*Removing unnecessary tests
